superset-frontend/packages/superset-ui-core/src/components/Label/reusable/DatasetTypeLabel.tsx

```diff
@@ -45,7 +45,16 @@ export const DatasetTypeLabel: React.FC<DatasetTypeLabelProps> = ({
   const labelType = datasetType === 'physical' ? 'primary' : 'default';
 
   return (
-    <Label icon={icon} type={labelType}>
+    <Label
+      icon={icon}
+      type={labelType}
+      style={{
+        color:
+          labelType === 'primary'
+            ? theme.colors.primary.dark2
+            : theme.colorPrimary,
```

Comment:
Is it possible to rename these? because the condition is `=== 'primary'` and then both colors are called primary, actually it's counterintuitive that i f`=== 'primary'` we use dark2 and when not primary we use `colorPrimary`?